### DISCUSSION TITLE
Non-official docker version available

### DISCUSSION THREAD

**Discussion by mp-se** [thorrak/fermentrack#581] [Show and tell]
There is a offical docker version in progress but I've created an alternative that is based on a single image. I've just tested the firmware flashing and bluetooth support so all the features should be working. 

The target for this image is a x86 platform, I run mine on a synology NAS both it has also been tested under Ubuntu. It must be run in privileged mode with host network in order for firmware flashing and bluetooth to work correctly. if you dont use these functions just run it in standard mode. Bluetooth support might require a Linux kernel 4.0 in order to work properly (depends on underlying file system). 

It can be downloaded from docker hub, search for "mpse2/fermentrack-docker" if you want to test it.

**Comment by thorrak** [comment]
I need to add instructions for installing your version to the documentation (and, ideally, create an install script for it). 

If you ever want to try to get GitHub actions set up for automatic generation/uploading, let me know. I can get you set up with contributor access to the repo to make it easier to maintain

**Comment by mp-se** [comment]
Sure, I would be glad to help with that. I still have some documentation for the project to create. How to you see that the installation would work ? I've only used the image on my Synology NAS and it's just a few clicks to download the image and setup the volumes. 

But if you are using docker on linux it would be easy to create a script that handle the build, setup and customize the docker-compose file. 

I've made it possible to run the image in both privileged and non privileged mode since I dont use bluetooth or do firmware upload from my NAS. I dont really like having docker images running as root. 

I will have a look at github actions, I've not really used them so far. I'm currently using the functions in docker hub to pull and create the image but I would be nice to do it in the other way. 

Ps. I love the Tilt simulator you found, works perfect for testing Ds.

**Comment by mp-se** [comment]
I looked at your installation script for docker and I believe that it would not be to difficult to adapt that to install my docker version as well.

I will give it a try to get it working on a raspberry pi.

**Comment by mp-se** [comment]
I've updated my build process to support the raspberry pi as well. Now there are images built for armv7, amd64 and i386. I've tested them on Raspberry PI 3B+, Synology NAS and Ubuntu desktop. Even the documentation has been updated with some better guides for installation.
